platform/api/oci/instance.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/oracle/bmcs-go-sdk"
+
+	"github.com/coreos/mantle/platform"
+)
+
+func (a *API) CreateInstance(name, userdata string) (*Machine, error) {
+	vcn, err := a.GetVCN("kola")
+	if err != nil {
+		return nil, err
+	}
+
+	subnet, err := a.getSubnetOnVCN(vcn.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := baremetal.LaunchInstanceOptions{
+		Metadata: map[string]string{
+			"user_data": base64.StdEncoding.EncodeToString([]byte(userdata)),
+		},
+		CreateVnicOptions: &baremetal.CreateVnicOptions{
+			AssignPublicIp: boolToPtr(true),
+			SubnetID:       subnet.ID,
+		},
+	}
+
+	inst, err := a.client.LaunchInstance(subnet.AvailabilityDomain, a.opts.CompartmentID, a.opts.Image, a.opts.Shape, subnet.ID, &opts)
+	if err != nil {
+		return nil, err
+	}
+
+	id := inst.ID
+
+	err = platform.WaitForActive(5*time.Minute, 10*time.Second, func() (bool, error) {
+		inst, err = a.client.GetInstance(id)
+		if err != nil {
+			return false, err
+		}
+
+		if inst.State != "RUNNING" {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		a.TerminateInstance(id)
+		return nil, fmt.Errorf("waiting for machine to become active: %v", err)
+	}
+
+	mach := Machine{}
```

Comment:
Just put `Name` and `ID` in the `Machine{}` initializer.  You could even move this to the end of the function and set all the fields at once.